### PR TITLE
Cart context and components

### PR DESCRIPTION
**Pull request by mrdigre**

### CODE REVIEW

File: src/app/cart/page.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use client'
+import Cart from "../components/Cart";
+
+const CartPage = () => {
+  return (
+    
+   <div>hi</div>
```
Comment (MiltonTulli) [inline src/app/cart/page.tsx:7]: WTF is this? Please remove if is unnecesary

**Review by MiltonTulli — CHANGES_REQUESTED** [review]

**Comment by mrdigre** [comment]
@MiltonTulli you broke the code.

File: src/app/components/CartButton.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+'use client'
+import React from 'react';
+import { useCart } from '@/app/contexts/CartContext';
+
+const CartButton = ({ product }) => {
+  const { addToCart } = useCart(); //context
+
+  const handleAddToCart = () => {
+    addToCart(product);
+    window.alert("Product added succesfully")
```
Comment (MiltonTulli) [inline src/app/components/CartButton.tsx:10]: You could implement something like https://www.npmjs.com/package/react-toastify for notifications

Comment (MiltonTulli) [reply src/app/components/CartButton.tsx:10]: Even the `addToCart` function from context could be the one that triggers the notification.

File: src/app/components/Cart.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                            const existingItem = aggregatedItems.find(uniqueItem => uniqueItem.id === item.id);
+
+                            if (existingItem) {
+                            existingItem.quantity += item.quantity;
+                            } else {
+                                aggregatedItems.push({ ...item });
+                        
+                            }
+
+                            return aggregatedItems;
```
Comment (MiltonTulli) [inline src/app/components/Cart.tsx:27]: This logic is to find duplicated values, right? I think you should handle this when adding product to cart, not when consuming items.

File: src/app/contexts/CartContext.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    setCartItems((prevItems) => prevItems.filter(item => item.id !== productId));
+  };
+  
+  const cartValue = {
+    cartItems,
+    addToCart,
+    removeFromCart
+  };
+
+  console.log("CartProvider - cartItems:", cartItems);
```
Comment (MiltonTulli) [inline src/app/contexts/CartContext.tsx:35]: You can remove log

**Review by MiltonTulli — COMMENTED** [review]
Just a few comments. Also run `npm run lint` and fix lint issues

**Pull request merged**